packages/app-staking/src/Overview/CurrentList.tsx

```diff
@@ -20,17 +22,66 @@ type Props = I18nProps & {
   recentlyOffline: RecentlyOfflineMap
 };
 
-class CurrentList extends React.PureComponent<Props> {
+type State = {
+  filter: ValidatorFilter,
+  filterOptions: Array<{ text: React.ReactNode, value: ValidatorFilter }>
+};
+const Wrapper = styled.div`
+  .filter {
+    display: flex;
+    justify-content: flex-end;
+    margin-bottom: 0.75rem;
+
+    > div {
+      max-width: 35rem;
+    }
+  }
+`;
+
+class CurrentList extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    const { t } = props;
+
+    this.state = {
+      filter: 'all',
+      filterOptions: [
+        { text: t('Show all validators and intentions'), value: 'all' },
+        { text: t('Show only my nominations'), value: 'iNominated' },
+        { text: t('Show only with nominators'), value: 'hasNominators' },
+        { text: t('Show only without nominators'), value: 'noNominators' },
+        { text: t('Show only with warnings'), value: 'hasWarnings' },
+        { text: t('Show only without warnings'), value: 'noWarnings' }
+      ]
+    };
```

Comment:
The `filterOptions` don't change so you could declare them as a const instead of part of the component state.

Reply:
I'm not really sure how to do it since I need  `t()` from props.
The only possible alternative  I can come up with is something like `this` 
```
class CurrentList extends React.PureComponent<Props, State> {
  state: State;
  filterOptions: FilterOptions;

  constructor (props: Props) {
    super(props);

    const { t } = props;

    this.state = {
      filter: 'all'
    };

    this.filterOptions = [
      { text: t('Show all validators and intentions'), value: 'all' },
      { text: t('Show only my nominations'), value: 'iNominated' },
      { text: t('Show only with nominators'), value: 'hasNominators' },
      { text: t('Show only without nominators'), value: 'noNominators' },
      { text: t('Show only with warnings'), value: 'hasWarnings' },
      { text: t('Show only without warnings'), value: 'noWarnings' }
    ]
  }
```
but I'm not declaring them as const this way.
¿How would you do it?

Reply:
Good point! And we're actually putting translated strings into the state in other components, so just leave it as it is.